bridges/snowbridge/pallets/system-v2/src/lib.rs

```diff
@@ -141,40 +139,36 @@ pub mod pallet {
 		/// Sends a command to the Gateway contract to instantiate a new agent contract representing
 		/// `origin`.
 		///
-		/// Fee required: Yes
-		///
-		/// - `origin`: Must be `Location` of a sibling parachain
+		/// - `location`: The location representing the agent
+		/// - `fee`: Ether to pay for the execution cost on Ethereum
 		#[pallet::call_index(1)]
 		#[pallet::weight(T::WeightInfo::create_agent())]
 		pub fn create_agent(
 			origin: OriginFor<T>,
 			location: Box<VersionedLocation>,
 			fee: u128,
 		) -> DispatchResult {
-			let _ = match origin.clone().into() {
-				Ok(RawOrigin::Root) => Ok(Here.into()),
-				_ => T::SiblingOrigin::ensure_origin(origin),
-			}?;
+			T::SiblingOrigin::ensure_origin(origin)?;
 
 			let origin_location: Location =
 				(*location).try_into().map_err(|_| Error::<T>::UnsupportedLocationVersion)?;
 
-			let ethereum_location = T::EthereumLocation::get();
 			// reanchor to Ethereum context
-			let location = origin_location
+			let ethereum_location = T::EthereumLocation::get();
+			let reanchored_location = origin_location
 				.clone()
 				.reanchored(&ethereum_location, &T::UniversalLocation::get())
 				.map_err(|_| Error::<T>::LocationConversionFailed)?;
 
-			let agent_id = agent_id_of::<T>(&location)?;
+			let agent_id = agent_id_of::<T>(&reanchored_location)?;
```

Comment:
I think you should move this location->agent conversion to a common helper function. I can already see that this code is duplicated between create_agent and register_token. Yet the implementations differ slightly.

Reply:
It’s just one line of code, so I prefer to inline it. 
https://github.com/vgeddes/polkadot-sdk/pull/6/commits/da0ef5d17a6c1a0c57bcacde6840c8205725185e

Reply:
Its not just the agent_id conversion - it includes the the reanchoring too.

There should be a single source of truth for reanchoring and conversion.